docs/configuration.md

```diff
@@ -7,6 +7,93 @@ nav_order: 10
 Modular has minimal configuration because of its philosophy. However there is a
 set of minimum configuration required.
 
+## Configuration File
+
+We allow a number of Modular behaviours to be configured via a dedicated Modular
+config file, `.modular.js`, located at the root of the repository.
+
+We support the following file names/formats:
+
+- `modular` property within your `package.json`
+- `.modular.js`
+- `.modularrc`
+- `.modularrc.json`
+- `.modularrc.yaml`
+- `.modularrc.yml`
+- `.modularrc.js`
+- `.modularrc.cjs`
+- `modular.config.js`
+- `modular.config.cjs`
+
+Example `.modular.js` file contents with all configurable attributes and their
+default values:
+
+```js
+module.exports = {
+  useModularEsbuild: false,
+  externalCdnTemplate: 'https://esm.sh/[name]@[version]',
+  externalBlockList: [],
+  externalAllowList: ['**'],
+  publicUrl: '',
+  generateSourceMap: true,
+};
+```
+
+### **useModularEsbuild**
+
+**Type**: `boolean`
+
+**Default**: `false` - Uses [Webpack](https://webpack.js.org/)
+
+Use esbuild instead of default Webpack. Only affects Views and ESM Views.
+
+### **externalCdnTemplate**
+
+**Type**: `string`
+
+**Default**: `https://esm.sh/[name]@[version]` - [esm.sh](https://esm.sh/)
+
+Template to resolve the URL used to fetch packages from a CDN. Defaults to
+esm.sh. Only applies to ESM Views.
+
+### **externalBlockList**
+
+**Type**: `string[]`
+
+**Default**: `[]` - No packages
+
+Packages that should be bundled and not fetched from a CDN. Avoid using this
```

Comment:
I think "Avoid using this unless absolutely necessary" is a bit vague. Can we give some clearer guidance?

Perhaps something like

```markdown
It is not recommended to change the default value, but this option exists to allow special handling of certain dependencies, e.g. `react`.
```

Reply:
Good point, I've now tried to add some details and linked the documentation page containing all details regarding CDN Limitations